pkg/kubelet/pod/pod_manager_test.go

```diff
@@ -37,131 +40,185 @@ func newTestManager() (*basicManager, *podtest.FakeMirrorClient) {
+	var (
+		mirrorPod = &v1.Pod{
+			ObjectMeta: metav1.ObjectMeta{
+				UID:       types.UID("mirror-pod-uid"),
+				Name:      "mirror-static-pod-name",
+				Namespace: metav1.NamespaceDefault,
+				Annotations: map[string]string{
+					kubetypes.ConfigSourceAnnotationKey: "api",
+					kubetypes.ConfigMirrorAnnotationKey: "mirror",
+				},
 			},
-		},
-	}
-	staticPod := &v1.Pod{
-		ObjectMeta: metav1.ObjectMeta{
-			UID:         "123456789",
-			Name:        "bar",
-			Namespace:   "default",
-			Annotations: map[string]string{kubetypes.ConfigSourceAnnotationKey: "file"},
-		},
-	}
-
-	expectedPods := []*v1.Pod{
-		{
+		}
+		staticPod = &v1.Pod{
 			ObjectMeta: metav1.ObjectMeta{
-				UID:         "999999999",
-				Name:        "taco",
-				Namespace:   "default",
-				Annotations: map[string]string{kubetypes.ConfigSourceAnnotationKey: "api"},
+				UID:         types.UID("static-pod-uid"),
+				Name:        "mirror-static-pod-name",
+				Namespace:   metav1.NamespaceDefault,
+				Annotations: map[string]string{kubetypes.ConfigSourceAnnotationKey: "file"},
 			},
+		}
+		normalPod = &v1.Pod{
+			ObjectMeta: metav1.ObjectMeta{
+				UID:       types.UID("normal-pod-uid"),
+				Name:      "normal-pod-name",
+				Namespace: metav1.NamespaceDefault,
+			},
+		}
+	)
+
+	testCase := []struct {
+		name         string
+		podList      []*v1.Pod
+		wantPod      *v1.Pod
+		expectPods   []*v1.Pod
+		expectGetPod *v1.Pod
+		expectUID    types.UID
+	}{
+		{
+			name:         "Get normal pod",
+			podList:      []*v1.Pod{mirrorPod, staticPod, normalPod},
+			wantPod:      normalPod,
+			expectPods:   []*v1.Pod{staticPod, normalPod},
+			expectGetPod: normalPod,
+			expectUID:    types.UID("static-pod-uid"),
+		},
+		{
+			name:         "Get static pod",
+			podList:      []*v1.Pod{mirrorPod, staticPod, normalPod},
+			wantPod:      staticPod,
+			expectPods:   []*v1.Pod{staticPod, normalPod},
+			expectGetPod: staticPod,
+			expectUID:    types.UID("static-pod-uid"),
 		},
-		staticPod,
-	}
-	updates := append(expectedPods, mirrorPod)
-	podManager, _ := newTestManager()
-	podManager.SetPods(updates)
-
-	// Tests that all regular pods are recorded correctly.
-	actualPods := podManager.GetPods()
-	if len(actualPods) != len(expectedPods) {
-		t.Errorf("expected %d pods, got %d pods; expected pods %#v, got pods %#v", len(expectedPods), len(actualPods),
-			expectedPods, actualPods)
 	}
-	for _, expected := range expectedPods {
-		found := false
-		for _, actual := range actualPods {
-			if actual.UID == expected.UID {
-				if !reflect.DeepEqual(&expected, &actual) {
-					t.Errorf("pod was recorded incorrectly. expect: %#v, got: %#v", expected, actual)
+	for _, test := range testCase {
+		t.Run(test.name, func(t *testing.T) {
+			podManager, _ := newTestManager()
+			podManager.SetPods(test.podList)
+			actualPods := podManager.GetPods()
+			sort.Slice(actualPods, func(i, j int) bool {
+				return actualPods[i].Name < actualPods[j].Name
+			})
```

Comment:
I think it would be better to keep this comment for readability.

```suggestion
			podManager, _ := newTestManager()
			podManager.SetPods(test.podList)

			// Tests that all regular pods, except for mirror pods, are recorded correctly.
			actualPods := podManager.GetPods()
			sort.Slice(actualPods, func(i, j int) bool {
				return actualPods[i].Name < actualPods[j].Name
			})
```

Reply:
fixed